### ISSUE TITLE
code in a variable block isn't linted

### ISSUE THREAD

**Issue by gshennessy** [sqlfluff/sqlfluff#5279]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

If I have a file a variable block, the code inside the variable block isn't linted.

```
select                  1;

do $$
declare datum integer;
begin
datum := (
      select                    1
);
end $$
```

### Expected Behaviour

The first line has the tabs replaced with spaces. I would expect the code inside the variable also be linted, aka, the tabs replaced with spaces.

### Observed Behaviour

While the tabs in the first line are replaced with spaces, the tabs inside the variable block are not.

### How to reproduce

Copy  the code I provide into a file, and use the config file I provided. Run sqlfluff lint file.sql, and it complains about line 1, but does not complain about lines 6 through 8. . run sqlfluff fix file.sql and it fixes line 1, but does not fix line 7.

### Dialect

postgres

### Version

[ariel:2 ~]$ sqlfluff --version
sqlfluff, version 2.3.2

### Configuration

[sqlfluff]

# Supported dialects https://docs.sqlfluff.com/en/stable/dialects.html
# Or run 'sqlfluff dialects'
dialect = postgres

# some of the dcdb sqitch scripts are larg
large_file_skip_char_limit = 0
large_file_skip_byte_limit = 0

# The standard max_line_length is 80 in line with the convention of
# other tools and several style guides. Many projects however prefer
# something a little longer.
# Set to zero or negative to disable checks.
[sqlfluff:indentation]
max_line_length = 120
tab_space_size = 4
indent_unit = space

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
Hi @gshennessy  - thanks for opening this issue.

Please see also #5488 

Postgres anonymous code block can be in any language according to the documentation, not just SQL or plpqsql: https://www.postgresql.org/docs/current/sql-do.html

Sqlfluff won't parse these blocks and treat them as literal strings, and therefore won't lint them. There is no roadmap to provide external linter hand-off and child execution to python or similar linters.
